devito/passes/clusters/blocking.py

```diff
@@ -157,3 +160,85 @@ def decompose(ispace, d, block_dims):
     directions.update({bd: ispace.directions[d] for bd in block_dims})
 
     return IterationSpace(intervals, sub_iterators, directions)
+
+
+class Skewing(Queue):
+
+    """
+    Rebuild clusters with skewed expressions and skewed IterationSpace.
+
+    Notes
+    -----
+    This transformation is applying Loop skewing to derive the
+    wavefront method of execution of nested loops.  Loop skewing is
+    a simple transformation of loop bounds and is combined with loop
+    interchanging to generate the wavefront [1]_.
+
+    .. [1] Wolfe, Michael. "Loops skewing: The wavefront method revisited."
+    International Journal of Parallel Programming 15.4 (1986): 279-293.
+
+    Examples:
+
+    .. code-block:: python
+
+        for i = 2, n-1
+            for j = 2, m-1
+                a[i,j] = (a[a-1,j] + a[i,j-1] + a[i+1,j] + a[i,j+1]) / 4
+            end for
+        end for
+
+        to
```

Comment:
"to" outside of the code-block, then open a new code-block?

also, the ::python doesn't make sense since you're not using python syntax....unless you drop the `end for` (at least)

Reply:
Sure.